libs/datasets/dataset_pointer.py

```diff
@@ -51,7 +53,32 @@ def save_dataset(self, dataset: DatasetBase) -> pathlib.Path:
         _logger.info("Successfully saved dataset", path=str(self.path))
         return self.path
 
-    def load_dataset(self) -> DatasetBase:
+    def load_dataset(
+        self, before: str = None, previous_commit: bool = False, commit: str = None
```

Comment:
instead of having a bunch of parameters how about a string you pass to `git rev-list -n` ? That supports dates, branches, HEAD^, HEAD^^ and more

Reply:
I do like the idea but don't quite have the energy right now to do it, would you be alright if I add this as a TODO and keep it how it is?

Reply:
No problem. Please add the comment.